pkg/manager-server/file_system_api.go

```diff
@@ -212,8 +212,15 @@ func (*FileSystem) run(cmd string) ([]byte, error) {
 		shellCmd.Stdout = &fsError.stdout
 		shellCmd.Stderr = &fsError.stderr
 		err := shellCmd.Run()
+
+		// Command timed out, return timeout message with FileSystemError
+		if ctx.Err() == context.DeadlineExceeded {
+			errStr := fmt.Sprintf("command timed out after %s seconds", timeoutString)
+			return []byte(errStr), fmt.Errorf("%s: %w", errStr, &fsError)
+		}
+
+		// Command failed, return FileSystemError with stdout and stderr
```

Comment:
Why did you pull this out of the 'if' below?

Reply:
Personal preference + consistency with the rest of the file. There's not a lot of blocks in this file, but if you look at `RegisterFileSystem()`, it puts comments before the block, not inside.